src/psycopmlutils/timeseriesflattener/flattened_dataset.py

```diff
@@ -299,33 +279,51 @@ def _cache_is_hit(
                 **kwargs_dict,
             )
 
-            n_to_generate = (
-                n_to_generate**1.5
-            )  # Increase n_to_generate by 1.5x each time to increase chance of non_fallback values
+            # Keep only values in generated_df which are not NaN
+            generated_df = generated_df[generated_df[full_col_str].notna()]
```

Comment:
Make this based on the fallback value instead?

Reply:
Comparison to fallback happens in the if statement in l. 292, so I don't think it matters here :-)